sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusReactorReceiver.java

```diff
@@ -362,7 +362,7 @@ private void cleanupWorkItems() {
         });
     }
 
-    private void completeWorkItem(String lockToken, Delivery delivery, MonoSink<Void> sink, Throwable error) {
+    private void completeWorkItem(String lockToken, Delivery delivery, MonoSink<Object> sink, Throwable error) {
```

Comment:
Why do we need it to be a MonoSink of Object? I thought you could cache to `.<Void>cache()`?

Reply:
the `.cache()` call turns the sink into a MonoSink<Object> from MonoSink<Void>.
```java
final Mono<Void> result = Mono.create(sink -> {
            workItem.start(sink);
            try {
                provider.getReactorDispatcher().invoke(() -> {
                    unsettled.disposition(deliveryState);
                    pendingUpdates.put(lockToken, workItem);
                });
            } catch (IOException error) {
                sink.error(new AmqpException(false, "updateDisposition failed while dispatching to Reactor.",
                    error, handler.getErrorContext(receiver)));
            }
        }).cache().then();
```

Reply:
You can cast the create operator to `Void` as shown below.

```java
final Mono<Void> result = Mono.<Void>create(sink -> {
            workItem.start(sink);
            try {
                provider.getReactorDispatcher().invoke(() -> {
                    unsettled.disposition(deliveryState);
                    pendingUpdates.put(lockToken, workItem);
                });
            } catch (IOException error) {
                sink.error(new AmqpException(false, "updateDisposition failed while dispatching to Reactor.",
                    error, handler.getErrorContext(receiver)));
            }
        }).cache();
```